### PR TITLE
 Add hash: IHash on Codec base

### PR DESCRIPTION
**Pull request by jacogr**
Builds on #1192

### CODE REVIEW

**Comment by jacogr** [comment]
Build is fixed on #1192, so will recover. (Once that is merged will re-base to master)

File: packages/types/src/primitive/Null.ts [hunk 1]
```diff
@@ -18,6 +22,13 @@ export default class Null implements Codec {
     return 0;
   }
 
+  /**
+   * @description returns a hash of the contents
+   */
+  public get hash (): IHash {
+    return createType('Hash', blake2AsU8a(this.toU8a(), 256));
```
Comment (amaurymartiny) [inline packages/types/src/primitive/Null.ts]: hmm, it's sometimes `createType('Hash',)`, sometimes `new U8a(...)`. For consistency I would just use createType everywhere

Comment (jacogr) [reply packages/types/src/primitive/Null.ts]: Can't use `createType` in codec without running into the issues of `extending from null`

i.e. in U8a -

- U8aFixed extends U8a
- U8a imports createType
- createType imports U8aFixed
- circular dep, all hell breaks loose :)

Comment (amaurymartiny) [reply packages/types/src/primitive/Null.ts]: ah, f, I see

**Pull request merged**